src/Cron/DayOfWeekField.php

```diff
@@ -36,7 +36,10 @@ public function isSatisfiedBy(\DateTime $date, $value)
             $tdate = clone $date;
             $tdate->setDate($currentYear, $currentMonth, $lastDayOfMonth);
             while ($tdate->format('w') != $weekday) {
-                $tdate->setDate($currentYear, $currentMonth, --$lastDayOfMonth);
+                $tdateClone = new \DateTime();
```

Comment:
For some reason the day of `$tdate` stays the same when using `setDate();`. This leads to an infinite loop since `$tdate->format('w')` is unchanged. This is not the most beautiful workaround, but it works.